lib/jxl/libjxl.pc.in

```diff
@@ -7,7 +7,7 @@ Name: libjxl
 Description: Loads and saves JPEG XL files
 Version: @JPEGXL_LIBRARY_VERSION@
 Requires.private: @JPEGXL_LIBRARY_REQUIRES@
-Libs: -L${libdir} -ljxl
-Libs.private: -lm
+Libs: -L${libdir} -l@JPEGXL_LIBRARY_MAIN@
+Libs.private: -l@JPEGXL_MATH_LIBRARY@
```

Comment:
Maybe move the `-l` inside the variable as well? Otherwise we'll get just the empty and unwanted `-l` here, no?

Reply:
Furthermore, the math library is not necessarily separate on all platforms. E.g. [LibTIFF uses the `find_package(CMath)` do detect this, and then appends only what is necessary](https://gitlab.com/libtiff/libtiff/-/blob/07f922106bd0985b65ed51763f0a79dc117f4d95/libtiff/CMakeLists.txt#L166).

Reply:
To avoid introducing the libtiff complexity, I suggest splitting this into two parts: one handling `jxl_static` for MSVC only, and the second leaving out `-lm` for _any_ WIN32 and APPLE

Reply:
Done as per your suggestion.

Reply:
LGTM, thanks!